DialogflowWrapper.ts

```diff
@@ -0,0 +1,103 @@
+import { IHttp, IHttpRequest } from '@rocket.chat/apps-engine/definition/accessors';
+import { createSign } from 'crypto';
+import { base64urlEncode } from './helper';
+
+export class DialogflowWrapper {
+
+    private clientEmail: string;
+    private privateKey: string;
+
+    constructor(clientEmail: string, privateKey: string) {
+        this.clientEmail = clientEmail;
+        this.privateKey = privateKey.trim().replace(/\\n/gm, '\n');
+    }
+
+    public async getAccessToken(http: IHttp) {
+
+        const authUrl = 'https://oauth2.googleapis.com/token';
+
+        const jwt = this.getJWT();
+
+        const httpRequestContent: IHttpRequest = {
+            headers: {
+                'Content-Type': 'application/x-www-form-urlencoded',
+            },
+            content: `grant_type=urn:ietf:params:oauth:grant-type:jwt-bearer&assertion=${jwt}`,
+        };
+
+        try {
+            const response = await http.post(authUrl, httpRequestContent);
+
+            const responseJSON = JSON.parse((response.content || '{}'));
+
+            if (responseJSON.access_token) {
+                const accessToken = responseJSON.access_token;
+                return accessToken;
+            } else {
+                if (responseJSON.error) {
+                    throw Error(`\
+                    ---------------------Error with Google Credentials-------------------\
+                    Details:- \
+                        Error Message:- ${responseJSON.error} \
+                        Error Description:- ${responseJSON.error_description}`);
+                }
+                throw Error('Error retrieving access token');
+            }
+
+        } catch (error) {
+            throw Error(error);
+        }
+    }
+
+    private getJWT() {
+        // request format
+        // {Base64url encoded header}.{Base64url encoded claim set}.{Base64url encoded signature}
+
+        const header = this.getJWTHeader();
+        const claimSet = this.getClaimSet();
+        const signature = this.getSignature(header, claimSet);
+        // combining all together to form jwt
+        const jwt = header + '.' + claimSet + '.' + signature;
+        return jwt;
+    }
+
+    // Forming the JWT header
+    private getJWTHeader() {
+        const jwtHeader = '{"alg":"RS256","typ":"JWT"}';
+        const b64uHeader = base64urlEncode(jwtHeader);
```

Comment:
```suggestion
        const b64uHeader = base64urlEncode('{"alg":"RS256","typ":"JWT"}');
```

Are you sure the function parameter is a string instead of an object?

Reply:
Yes. It needs to be a string. Bcs, later on we pass this to [`encodeURIComponent`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/encodeURIComponent) which expects string.
REF: https://github.com/RocketChat/Apps.Dialogflow/blob/c94ce724504c200ed7ee7988c46da270e58d745b/helper.ts#L4

Reply:
So, this is a decision of the target method, not the method where you're calling it, right?